### PR TITLE
Improve warnings before perf attribution

### PR DESCRIPTION
**Pull request by yankees714**
- Adds a new check for algorithms whose turnover is too high
- Updates language for the warning emitted when asses are missing from the risk factor loadings

### CODE REVIEW

File: pyfolio/perf_attrib.py [hunk 1]
```diff
@@ -114,14 +136,24 @@ def perf_attrib(returns, positions, factor_returns, factor_loadings,
 
     if len(missing_stocks) > 0:
 
-        warnings.warn("Could not find factor loadings for the following "
```
Comment (vikram-narayan) [inline pyfolio/perf_attrib.py]: I think you'll need to update the tests as well with the new language?

File: pyfolio/perf_attrib.py [hunk 2]
```diff
@@ -114,14 +136,24 @@ def perf_attrib(returns, positions, factor_returns, factor_loadings,
...
+            "accounted for in performance attribution.\n"
+            "\n"
+            "The following assets were missing factor loadings: {}. "
+            "Ignoring for exposure calculation and performance attribution. "
+            "Ratio of assets missing: {}. Average allocation of missing "
+            "assets:\n"
+            "\n"
+            "{}.\n"
+        ).format(
+            list(missing_stocks),
```
Comment (vikram-narayan) [inline pyfolio/perf_attrib.py]: should we also truncate this list if it's too long?

File: pyfolio/perf_attrib.py [hunk 3]
```diff
@@ -114,14 +136,24 @@ def perf_attrib(returns, positions, factor_returns, factor_loadings,
...
-                      "stocks: {}. Ignoring for exposure calculation and "
-                      "performance attribution. Coverage ratio: {}/{}. "
-                      "Average allocation of missing stocks: {} "
-                      .format(list(missing_stocks),
-                              num_stocks_covered,
-                              num_stocks,
-                              positions[missing_stocks].mean()))
+        missing_stocks_warning_msg = (
+            "Could not determine risk exposures for some of this algorithm's "
+            "positions. PnL from the missing assets will not be properly "
```
[the remark this answers is not in the document]
Comment (yankees714) [reply pyfolio/perf_attrib.py]: @abhijeetkalyan thoughts on this?

File: pyfolio/perf_attrib.py (hunk 2 above)
Comment (yankees714) [reply pyfolio/perf_attrib.py]: Planning to do this in a follow-up PR

Comment (twiecki) [reply pyfolio/perf_attrib.py]: Same with missing dates, don't want to list all of them but just the period.

Comment (abhijeetkalyan) [reply pyfolio/perf_attrib.py]: Yup - @twiecki that's another issue that we're tracking

File: pyfolio/perf_attrib.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (abhijeetkalyan) [reply pyfolio/perf_attrib.py]: Agreed, returns is clearer

**Comment by yankees714** [comment]
@vikram-narayan Should be good for another look

**Comment by yankees714** [comment]
Failing travis job looked like #454, hopefully the next run will be fine

**Pull request merged**